setup.sh

```diff
@@ -1,59 +1,53 @@
 #!/bin/bash
 
-grep_ver="grep -o [0-9.]*"
-grep_perm_denied="grep -o Permission"
-grep_success="grep -o Successfully"
-
-check_install_pkg() {
-    pip_tool="pip$1"
+check_install_module() {
+    module="$1"
     pkg="$2"
-    #Check python-$pkg version, if not available install
-    pkg_ver=$($pip_tool show $pkg 2>&1| grep --line-buffered "^Version" | $grep_ver) 
-    if [ -n "$pkg_ver" ]; then
-        echo "$pkg version - $pkg_ver found"
-    else
-        echo "$pkg not found. Installing $pkg..."
-        install_output=$($pip_tool install $pkg 2>&1)
-        if [ -n "$(echo $install_output | $grep_perm_denied)" ]; then 
-            install_output=$($pip_tool install --user $pkg 2>&1)
-
-            if [ -n "$(echo $install_output | $grep_success)" ]; then 
-            pkg_ver=$($pip_tool show $pkg | grep --line-buffered "^Version"  | $grep_ver) 
-            echo "Successfully installed $pkg version $pkg_ver"
-            fi
-        elif [ -n "$(echo $install_output | $grep_success)" ]; then
-            pkg_ver=$($pip_tool show $pkg | grep --line-buffered "^Version"  | $grep_ver) 
-            echo "Successfully installed $pkg version $pkg_ver"
-        else 
-            echo "Unknown error installing $pkg"
-            echo $install_output
+    python3 -c "import $module" 2>&1
+    if [ $? -eq 0 ]; then
+        echo "$module already installed."
+        return
+    fi
+    echo "Installing $pkg"
+    sudo apt-get install $pkg
+    if [ $? -ne 0 ]; then
+        echo "Installing $pkg failed.\nAbort."
+        exit 1
+    fi
+}
+check_install_python3(){
+    if [ $(which python3) ]; then
+        python3_ver=$(python3 --version)
+        echo "$python3_ver already installed."
+        return
+    fi
+    echo "python3 not found. Installing python3..."
+    sudo apt-get install python3
+    if [ $? -ne 0 ]; then
+        echo "Installing python3 failed."
+        exit 1
+    fi
+}
+check_install_sphinx(){
+    if ! sphinx-build --v 2>/dev/null; then
+        if ls ~/.local/bin/sphinx-build 2>/dev/null; then
```

Comment:
ok. alternatively use built-in file-check:
```suggestion
        if ! [ -x ~/.local/bin/sphinx-build ]; then
```